src/object/cli_obj.c

```diff
@@ -345,13 +346,14 @@ obj_layout_create(struct dc_object *obj, bool refresh)
 
 	obj->cob_shards_nr = layout->ol_nr;
 	obj->cob_grp_size = layout->ol_grp_size;
+	old = obj->cob_grp_nr;
+	obj->cob_grp_nr = obj->cob_shards_nr / obj->cob_grp_size;
 
 	if (obj->cob_grp_size > 1 && srv_io_mode == DIM_DTX_FULL_ENABLED &&
-	    obj->cob_grp_nr < obj->cob_shards_nr / obj->cob_grp_size) {
+	    old < obj->cob_grp_nr) {
 		if (obj->cob_time_fetch_leader != NULL)
 			D_FREE(obj->cob_time_fetch_leader);
 
-		obj->cob_grp_nr = obj->cob_shards_nr / obj->cob_grp_size;
```

Comment:
seems the logic is same as before?...

Reply:
This can be removed.

Reply:
Originally, the obj::cob_grp_nr would be set conditionally. For example, if the object is non-replicated one, then its obj->cob_grp_nr would not be set, so always was zero, but that is incorrect.

Reply:
This patch makes the assignment of "obj->cob_grp_nr" unconditionally.

Reply:
ah, obj->cob_grp_size > 1 excluded the non-replica case, thx